File: CPP/ByRound/1698/1698_D_Fixed_Point_Guessing.cpp
// Problem: CF 1698 D - Fixed Point Guessing
// https://codeforces.com/contest/1698/problem/D

/*
Algorithm: Binary Search on the answer
Purpose: Find the fixed point (element that did not change position) in an array that was created by swapping pairs of elements from the original sorted array [1,2,...,n].
The key insight is that we can determine if an element at position i is a fixed point by querying the prefix [1..i] and checking how many elements out of those are <= i. 
If the count of such elements is odd, then i is not a fixed point (because in a correctly swapped array with one element being fixed, if we fix an element at position i, 
then the prefix [1..i] should contain exactly i elements that are <= i from a fully sorted sequence. Any deviation from the expected pattern indicates a mismatch).
Time Complexity: O(log n) per test case due to binary search
Space Complexity: O(1) - only using a few variables for computation
*/
#include <iostream>
using namespace std;
int main() {
  int t, n;
  cin >> t;
  while (t--) {
    cin >> n;
    int l = 1, r = n, ans;
    while (l <= r) {
      int mid = (l + r) / 2, cnt = 0, a;
      cout << "? 1 " << mid << endl;  // Query prefix [1..mid]
      for (int i = 1; i <= mid; i++)
        cin >> a, cnt += a <= mid;  // Count elements in [1..mid] that are <= mid
      if (cnt % 2)  // If count is odd, the element at mid is not fixed (since it should be even in sorted order)
        r = mid - 1, ans = mid;  // Move left and store mid as potential answer
      else
        l = mid + 1;  // If even, move right
    }
    cout << "! " << ans << endl;  // Output the fixed point
  }
  return 0;
}


// https://github.com/VaHiX/CodeForces/

File: Python/ByTier/D/1698_D_Fixed_Point_Guessing.py
# Problem: CF 1698 D - Fixed Point Guessing
# https://codeforces.com/contest/1698/problem/D

"""
Algorithm: Binary Search on the position of the fixed point
Time Complexity: O(log n) per test case, since we perform binary search on the array indices and each query takes O(n) time to process (but the number of queries is bounded by log n).
Space Complexity: O(1), only using a few variables for tracking indices and processing input/output.

The problem involves finding a fixed point in an array where half of the elements have been swapped in pairs,
and exactly one element remains in its original position. The key insight is that we can use binary search
to locate this fixed point by querying subarrays and checking how many elements of the left half are present.

We maintain a range [l, r] and at each step query the left half of the range. If the count of elements from the left half
in the sorted subarray is odd, the fixed point lies to the left (including mid). Otherwise, it lies to the right.
"""

for t in range(int(input())):
    n = int(input())
    l = 1
    r = n
    while l < r:
        m = l + (r - l) // 2  # Midpoint of the current search range
        print("?", l, m, flush=True)  # Query the left half of the current range
        # Read the sorted subarray and count how many elements are in [l, m]
        sorted_subarray = [int(x) for x in input().split()]
        count = sum(l <= i <= m for i in sorted_subarray)
        if count % 2:  # If count is odd, fixed point is in the left half
            r = m
        else:  # If count is even, fixed point is in the right half
            l = m + 1
    print("!", l, flush=True)  # Output the fixed point


# https://github.com/VaHiX/CodeForces/